[Merged by Bors] - chore: remove March and April 2025 deprecated declarations

Parcly-Taxel opened the pull request:
I undeprecate `Nat.Ico_succ_right_eq_insert_Ico`, since it is a natural counterpart to the nondeprecated `Nat.Ico_succ_left_eq_erase_Ico`.

bryangingechen commented:
> https://github.com/leanprover-community/mathlib4/pull/30901 was a mistake. The remove_deprecated_decls action did not fire on 15 December.

Do you still think my fix in #30901 was a mistake? I'd like to know if you found something wrong with it.

jcommelin commented:
@Parcly-Taxel Can you confirm that this PR was created using the tool?
Of did you use a custom tool?

Parcly-Taxel committed:
automated

import Mathlib
import Archive
import Counterexamples

-- #clear_deprecations "2025-01-01" "2025-04-30" really

Parcly-Taxel committed:
manual

Parcly-Taxel commented:
> Can you confirm that this PR was created using the tool? Of did you use a custom tool?

I ran, as explained in the first commit,
```lean
import Mathlib
import Archive
import Counterexamples

#clear_deprecations "2025-01-01" "2025-04-30" really
```
and then manually cleared the errors in the second commit.

j-loreaux reviewed (commented):
I'm merging this because there's no reason to have them around anymore, but @bryangingechen it would be good to know why the automation isn't firing.

bors merge
